File: electives/apm/lab/lista-1/ex-2/java/app/src/main/java/com/example/ordnung/ButtonMatrix.java
package com.example.ordnung;

import java.util.Random;

public enum ButtonMatrix {
    NW,
    N,
    NE,
    W,
    C,
    E,
    SW,
    S,
    SE;

    private static final ButtonMatrix[] VALUES = values();
    private static final int SIZE = VALUES.length;
    private static final Random RANDOM = new Random();

    public static ButtonMatrix GetRandom()  {
        return VALUES[RANDOM.nextInt(SIZE)];
    }

}

File: electives/apm/lab/lista-1/ex-2/kotlin/app/src/main/java/com/example/ordnung/ButtonMatrix.kt
package com.example.ordnung

import java.util.*

enum class ButtonMatrix {
    NW, N, NE, W, C, E, SW, S, SE;

    companion object {
        private val VALUES = values()
        private val SIZE = VALUES.size
        private val RANDOM = Random()
        fun GetRandom(): ButtonMatrix {
            return VALUES[RANDOM.nextInt(
                SIZE
            )]
        }
    }
}
